scripts/config.py

```diff
@@ -468,141 +504,124 @@ def _get_compatible_GCC(self, nvcc_version):
-
-                nvccDir = whereis(self.configDict["NVCC"])
-                if not checkCudaLib(cudaLib) and nvccDir:
-                    # Looking for Cuda libs under active nvcc.
-                    cudaLib = searchCudaLib(
-                        os.path.dirname(nvccDir), cudaLib, False)
-
-                if not checkCudaLib(cudaLib):
-                    # Looking for Cuda libs in user root libs.
-                    cudaLib = searchCudaLib('/usr', cudaLib, self.ask)
-
-                if checkCudaLib(cudaLib):
-                    self.configDict["NVCC_LINKFLAGS"] = ("-L%s" % cudaLib +
-                                                         " -L%s/stubs" % cudaLib)  # nvidia-ml is in stubs folder
-                    self.environment.update(LD_LIBRARY_PATH=cudaLib)
-                    self.environment.update(LD_LIBRARY_PATH=cudaLib+"/stubs")
-                else:
-                    print(yellow("WARNING: system libraries for CUDA not found!\n"
-                                 "         If cuda code is not compiling, "
-                                 "please, find 'libcudart.so' and manually add\n"
-                                 "         the containing folder (e.g. '/my/cuda/lib') at %s\n"
-                                 " > NVCC_LINKFLAGS = -L/my/cuda/lib -L/my/cuda/lib/stubs\n"
-                                 "         If the problem persist, set 'CUDA=False' before "
-                                 "compiling to skip cuda compilation."
-                                 % (Config.FILE_NAME)))
+    def _join_with_prefix(self, collection, prefix):
+        return ' '.join([prefix + i for i in collection if i])
+
+    def _set_nvcc(self):
+        if not self.is_empty(Config.OPT_NVCC):
+            return True
+
+        nvcc_loc_candidates = {os.environ.get('XMIPP_CUDA_BIN', ''),
+                               os.environ.get('CUDA_BIN', ''),
+                               '/usr/local/cuda/bin',
+                               '/usr/local/cuda*/bin'} #if there is no -ln on /usr/local/cuda ?
+        nvcc_loc = find('nvcc', nvcc_loc_candidates)
+        if not nvcc_loc:
+            print(yellow('nvcc not found (searched in %s). If you want to '
+                         'enable CUDA, add \'nvcc\' to PATH. %s' % (
+                nvcc_loc_candidates, self._get_help_msg())))
+            return False
+        print(green('nvcc found in ' + nvcc_loc))
+        self._set(Config.OPT_NVCC, nvcc_loc)
+        return True
+
+    def _set_nvcc_cxx(self, nvcc_version):
+        if not self.is_empty(Config.OPT_CXX_CUDA):
+            return True
+        candidates = self._get_compatible_GCC(nvcc_version)
+        prg = find_newest('g++', candidates,  False)
+        if not prg:# searching a g++ for devToolSet on CentOS
+            if str(self._get_GCC_version('g++')[0]) in candidates:
+                prg = whereis('g++', True)
+                if not prg:
```

Comment:
can it happen that you get `g++` versino (line 534) but then it's not found by `whereis`, i.e. is this condition useful?